x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -880,4 +809,24 @@ public static boolean isPlatinumOrTrialOperationMode(final OperationMode operati
     public synchronized XPackLicenseState copyCurrentLicenseState() {
         return new XPackLicenseState(this);
     }
+
+    private synchronized boolean checkSecurityEnabled() {
+        return isSecurityEnabled(status.mode, isSecurityExplicitlyEnabled, isSecurityEnabled);
+    }
+
+    private synchronized boolean checkMinimumLicense(
+        OperationMode minimumMode, boolean needSecurity, boolean needActive, boolean allowTrial) {
```

Comment:
The `allowTrial` argument is always true for all usages. It can be dropped unless anyone can foresee some future usage.

Reply:
The semantic of isTransport**Required** method is a bit different from the isXxx**Allowed** methods. So I can see trial could be handled differently. With that being said, I agree to keep the argument for 1) future proof; 2) consistent with pre-refactor logic

Reply:
> The semantic of isTransportRequired method is a bit different from the isXxxAllowed methods. So I can see trial could be handled differently.

agreed, I was just trying to make a point that it is not unheard of to have the need to treat trial differently.